dlup/data/transforms.py

```diff
@@ -79,12 +80,27 @@ def convert_annotations(
             points[curr_annotation.label] += tuple(coords)
             continue
 
-        if isinstance(curr_annotation, dlup.annotations.Polygon) and curr_annotation.type == AnnotationType.BOX:
+        if (
+            isinstance(curr_annotation, dlup.annotations.Polygon)
+            and curr_annotation.annotation_type == AnnotationType.BOX
+        ):
             min_x, min_y, max_x, max_y = curr_annotation.bounds
             boxes[curr_annotation.label].append(((int(min_x), int(min_y)), (int(max_x - min_x), int(max_y - min_y))))
             continue
 
-        if roi_name and curr_annotation.label == roi_name:
+        index_value: int
+        if curr_annotation.label != roi_name:
+            if index_map:
+                if curr_annotation.label not in index_map:
+                    raise ValueError(f"Label {curr_annotation.label} is not in the index map {index_map}")
+                index_value = index_map[curr_annotation.label]
+
+            else:
+                if curr_annotation.z_index is None and curr_annotation.label != roi_name:
+                    raise ValueError(f"Label {curr_annotation.label} has no z_index and no index_map is provided.")
+                assert curr_annotation.z_index
+                index_value = curr_annotation.z_index + 1
```

Comment:
Is it appropriate to use z-index as an index value for the mask? Also, if a user has specified a specific z-index, should we add one to it? This is a DLUP-specific implementation, where normally index zero is reserved for "background". It would make more sense to use the z-index as is, if we use it at all.

Reply:
No, I've removed it.